### PR TITLE
fix(@desktop/wallet): SendModal: Move logic to nim Part 2, parsing recipient address string to separate the chain prefixes

### LINKED ISSUE #12149: Move parsing user entered preferred chains logic to nim and remore code not required in case of accounts/saved addresses and preferred chains is already known

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fix(@desktop/wallet): SendModal: Move logic to nim Part 2, parsing recipient address string to separate the chain prefixes
fixes #12149

### What does the PR do

Move logic for parsing recipient text to nim side

### Affected areas

SendModal

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

### CODE REVIEW

File: src/app/modules/main/wallet_section/send/module.nim [hunk 1]
```diff
@@ -334,3 +334,26 @@ method getCollectiblesModel*(self: Module): collectibles.Model =
 method getNestedCollectiblesModel*(self: Module): nested_collectibles.Model =
   return self.nestedCollectiblesModel
 
+method splitAndFormatAddressPrefix*(self: Module, text : string, updateInStore: bool): string {.slot.} =
+  var tempPreferredChains: seq[int] = @[]
```
Comment (saledjenic) [inline src/app/modules/main/wallet_section/send/module.nim:338]: @Khushboo-dev-cpp you don't need `= @[]` it's already empty.

File: src/app/modules/main/wallet_section/send/module.nim [hunk 2]
```diff
@@ -334,3 +334,26 @@ method getCollectiblesModel*(self: Module): collectibles.Model =
...
+        tempPreferredChains.add(self.view.getNetworkChainId(word))
+        editedText = editedText & "<span style='color: " & chainColor & "'>" & word & "</span>" & ":"
+
+  if updateInStore:
+    if not chainFound:
+      self.view.updateRoutePreferredChains(self.view.getLayer1NetworkChainId())
+    else:
+      self.view.updateRoutePreferredChains(tempPreferredChains.join(":"))
+
+  editedText = editedText & "</a></p>"
```
Comment (saledjenic) [inline src/app/modules/main/wallet_section/send/module.nim:358]: @Khushboo-dev-cpp within this function I cannot figure out where are starting `<a>` and `<p>` tags (see only closing tags)?

File: src/app/modules/main/wallet_section/send/view.nim [hunk 3]
```diff
@@ -242,3 +240,30 @@ QtObject:
...
+    var address = ""
+    for word in plainText(text).split(':'):
+      if word.startsWith("0x"):
+        address = word
+    return address
+
+  proc getShortChainIds(self: View, chainShortNames : string): string {.slot.} =
+    if chainShortNames.isEmptyOrWhitespace():
+      return ""
+    var preferredChains: seq[int] = @[]
```
Comment (saledjenic) [inline src/app/modules/main/wallet_section/send/view.nim:266]: @Khushboo-dev-cpp the same as above, it's already empty.

File: src/app/modules/main/wallet_section/send/view.nim [hunk 4]
```diff
@@ -242,3 +240,30 @@ QtObject:
...
+    return self.fromNetworksModel.getNetworkChainId(shortName)
+
+  proc splitAndFormatAddressPrefix(self: View, text : string, updateInStore: bool): string {.slot.} =
+    return self.delegate.splitAndFormatAddressPrefix(text, updateInStore)
+
+  proc getAddressFromFormattedString(self: View, text : string): string {.slot.} =
+    var address = ""
+    for word in plainText(text).split(':'):
+      if word.startsWith("0x"):
+        address = word
```
Comment (saledjenic) [inline src/app/modules/main/wallet_section/send/view.nim:260]: @Khushboo-dev-cpp add `break` here.
Also in 100% cases the address will be the last element in the array that is the result of `split(':')`, thus it's much better to iterate from the last towards the first element.

File: src/app/modules/main/wallet_section/send/module.nim (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply src/app/modules/main/wallet_section/send/module.nim:358]: you are right :D
But it works somehow lol, adding it now

**Review by alaibe — APPROVED** [review]

File: src/app/modules/main/wallet_section/send/view.nim [hunk 5]
```diff
@@ -242,3 +240,32 @@ QtObject:
...
+
+  proc splitAndFormatAddressPrefix(self: View, text : string, updateInStore: bool): string {.slot.} =
+    return self.delegate.splitAndFormatAddressPrefix(text, updateInStore)
+
+  proc getAddressFromFormattedString(self: View, text : string): string {.slot.} =
+    var address = ""
+    var   splitWords = plainText(text).split(':')
+    for i in countdown(splitWords.len-1, 0):
+      if splitWords[i].startsWith("0x"):
+        address = splitWords[i]
```
Comment (saledjenic) [inline src/app/modules/main/wallet_section/send/view.nim:261]: `return splitWords[i]`

**Review by saledjenic — APPROVED** [review]

**Pull request merged**